features/admin.extensions.v2/components/administrators/api/organization.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AsgardeoSPAClient, HttpClientInstance } from "@asgardeo/auth-react";
+import { HttpMethods } from "@wso2is/core/models";
+import { AxiosError, AxiosRequestConfig, AxiosResponse } from "axios";
+import useRequest, {
+    RequestConfigInterface,
+    RequestErrorInterface,
+    RequestResultInterface,
+    SWRConfig
+} from "../../../../admin.core.v1/hooks/use-request";
+import { store } from "../../../../admin.core.v1/store";
+import { OrganizationInterface } from "../../../../admin.extensions.v1/components/administrators/models";
+
+/**
+ * Initialize an axios Http client.
+ *
+ */
+const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance().
+    httpRequest.bind(AsgardeoSPAClient.getInstance());
+
+/**
+ * Hook to get enterprise login enable config.
+ *
+ * @param organization - Organization name.
+ * @param revalidateIfStale - Revalidate if stale.
+ *
+ * @returns the organization config.
+ */
+export const useOrganizationConfigV2 =
+    <Data = OrganizationInterface, Error = RequestErrorInterface>
+    (
+        organization: string, requestOptions: SWRConfig<Data, Error>,
+        shouldFetch: boolean = true
+    ) : RequestResultInterface<Data, Error> => {
+        const requestConfig: RequestConfigInterface = {
+            headers: {
+                "Content-Type": "application/json"
+            },
+            method: HttpMethods.GET,
+            url: store.getState().config.endpoints.organizationEndpointV2.replace("{organization}", organization)
+        };
+
+        const { data, error, isValidating, mutate } = useRequest<Data, Error>(
+            shouldFetch ? requestConfig : null, requestOptions);
+
+        return {
+            data,
+            error: error,
+            isLoading: !error && !data,
+            isValidating,
+            mutate: mutate
+        };
+    };
+
+/**
+ * Hook to update enterprise login enable config.
+ *
+ * @param isEnterpriseLoginEnabled - Enterpriselogin is enabled/disabled.
+ *
+ * @returns a promise containing the response.
+ */
+export const updateOrganizationConfigV2 = (isEnterpriseLoginEnabled: OrganizationInterface): Promise<any> => {
```

Comment:
move this to a new file. Keep only API call in a file, also refrain from using any